cache-filter/src/filter/http.rs

```diff
@@ -83,7 +85,26 @@ impl HttpContext for CacheFilter {
 
         self.cache_key = CacheKey::from(&request_data.service_id, &request_data.app_id);
         self.req_data = request_data.clone();
-        match get_application_from_cache(&self.cache_key.as_string()) {
+
+        if let AppIdentifier::UserKey(ref user_key) = request_data.app_id {
+            match get_app_id_from_cache(&user_key) {
+                Ok(app_id) => {
+                    request_data.app_id = AppIdentifier::from(app_id);
+                    self.req_data.app_id = request_data.app_id.clone();
+                    self.cache_key.set_app_id(&request_data.app_id);
+                }
+                Err(e) => {
+                    info!(
+                        "ctxt {}: failed to map user_key to app_id: {:?}",
+                        context_id, e
+                    );
+                    // TODO: avoid multiple calls for identical requests
+                    return do_auth_call(self, self, &request_data);
```

Comment:
This case will happen when we don't know about a specific user key - it might or might not exist, so perhaps it is valid once we fetch the data from an auth call. I'd lower the logging to debug level.